cratedb_toolkit/cfr/jobstats.py

```diff
@@ -42,13 +43,43 @@
 report_cursor: t.Any
 last_scrape: int
 interval: float
-
-
-def boot(address: DatabaseAddress, report_address: t.Optional[DatabaseAddress] = None):
+anonymize_sql: bool = False
+deanonymize_sql: bool = False  # Added global flag for deanonymization
+decoder_dict_path: str
+
+
+def boot(
+    address: DatabaseAddress,
+    report_address: t.Optional[DatabaseAddress] = None,
+    anonymize_statements: bool = False,
+    decoder_dict_file: t.Optional[str] = None,
+    deanonymize_statements: bool = False,
+):
     # TODO: Refactor to non-global variables.
-    global stmt_log_table, last_exec_table, cursor, report_cursor, last_scrape, interval
+    global \
+        stmt_log_table, \
+        last_exec_table, \
+        cursor, \
+        report_cursor, \
+        last_scrape, \
+        interval, \
+        anonymize_sql, \
+        deanonymize_sql, \
+        decoder_dict_path
+    anonymize_sql = anonymize_statements
+    deanonymize_sql = deanonymize_statements
+    decoder_dict_path = decoder_dict_file
     schema = address.schema or "stats"
 
+    if anonymize_sql and decoder_dict_path:
+        logger.info(f"SQL anonymization is enabled, using dictionary: {decoder_dict_path}")
+
+    if deanonymize_sql and decoder_dict_path:
+        logger.info(f"SQL deanonymization is enabled, using dictionary: {decoder_dict_path}")
+
+    if decoder_dict_file is None:
+        raise ValueError("Could not pick up decoder file")
```

Comment:
Thanks for adding those guards in order to satisfy the mypy type checker. If you will move lines 80-81 to before line 71, it will signal success.